src/udp/udp.c

```diff
@@ -372,6 +373,49 @@ int udp_listen(struct udp_sock **usp, const struct sa *local,
 }
 
 
+/**
+ * Create and listen on a UDP Socket within a port range
+ *
+ * @param usp      Pointer to returned UDP Socket
+ * @param local    Local network address
+ * @param min_port Minimum port number
+ * @param max_port Maximum port number
+ * @param rh       Receive handler
+ * @param arg      Handler argument
+ * @return 0 if success, otherwise errorcode
+ */
+int udp_listen_range(struct udp_sock **usp, const struct sa *local,
+		     uint16_t min_port, uint16_t max_port,
+		     udp_recv_h *rh, void *arg)
+{
+	struct sa laddr;
+	int tries = 64;
+	int err = 0;
+
+	sa_cpy(&laddr, local);
+
+	/* try hard */
+	while (tries--) {
+		struct udp_sock *us;
+		uint16_t port;
+
+		port = (min_port + (rand_u16() % (max_port + 1 - min_port)));
```

Comment:
Perhaps the port should also be even, if used for RTP applications ?

Reply:
The goal of this PR is to allow odd RTP ports for the multicast feature, where RTCP  commonly is not applicable because of the scaling problems.

Indeed, RFC-3550 Section 11 tells that RTP ports SHOULD be odd in combination with RTCP. First, there is no MUST and it doesn't state anything for RTP only streams.

Thus, we suggest to have one API function `rtp_listen(..., min_port, max_port, ...)` that    opens odd-even port pairs for RTP+RTCP like it was the case. And open any odd or even port in the given range if RTP only is wanted.

An alternative would be to add another API function.